src/components/flyout/flyout.styles.ts

```diff
@@ -0,0 +1,282 @@
+      }
+      &:focus {
+        outline: none;
+      }
+      ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+        ${logicalCSS('max-width', '90vw')}
+      }
+    `,
+
+    // Flyout Sizes
+    // Note: Dashes are used because s, m, l are size values for multiple props
+    'flyoutSize--s': css`
+      &.euiFlyout--maxWidth-default {
+        ${logicalCSS('max-width', `${flyoutSizes.s.max}px`)}
+      }
+      ${euiBreakpoint(euiThemeContext, ['m', 'xl'])} {
+        ${logicalCSS('min-width', `${flyoutSizes.s.min}px`)}
+        ${logicalCSS('width', flyoutSizes.s.width)}
+      }
+      ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+        ${logicalCSS('min-width', 0)}
+        ${logicalCSS('width', `${flyoutSizes.s.min}px`)}
+      }
+    `,
+    'flyoutSize--m': css`
+      &.euiFlyout--maxWidth-default {
+        ${logicalCSS('max-width', `${flyoutSizes.m.max}px`)}
+      }
+      ${euiBreakpoint(euiThemeContext, ['m', 'xl'])} {
+        ${logicalCSS('min-width', `${flyoutSizes.m.min}px`)}
+        ${logicalCSS('width', flyoutSizes.m.width)}
+      }
+      ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+        ${logicalCSS('min-width', 0)}
+        ${logicalCSS('width', `${flyoutSizes.m.min}px`)}
+      }
+    `,
+    'flyoutSize--l': css`
+      &.euiFlyout--maxWidth-default {
+        ${logicalCSS('max-width', `${flyoutSizes.l.max}px`)}
+      }
+      ${euiBreakpoint(euiThemeContext, ['m', 'xl'])} {
+        ${logicalCSS('min-width', `${flyoutSizes.l.min}px`)}
+        ${logicalCSS('width', flyoutSizes.l.width)}
+      }
+      ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+        ${logicalCSS('min-width', 0)}
+        ${logicalCSS('width', `${flyoutSizes.l.min}px`)}
+      }
+    `,
+
+    // Side
+    right: css``,
+    left: css`
+      ${logicalCSS('border-right', euiTheme.border.thin)}
+      ${logicalCSS('border-left', 'none')}
+      ${logicalCSS('left', 0)}
+      ${logicalCSS('right', 'auto')}
+      clip-path: polygon(0 0, 150% 0, 150% 100%, 0 100%);
+      ${euiCanAnimate} {
+        animation: ${euiFlyoutLeft};
+      }
+    `,
+
+    // Type
+    overlay: css``,
+    push: css`
+      box-shadow: none;
+      clip-path: none;
+      animation-duration: 0s !important; // Don't animate on loading a docked nav
+      ${logicalCSS('border-left', euiTheme.border.thick)}
+      // Make sure the header shadows are above
+      z-index: ${Number(euiTheme.levels.header) - 1};
+    `,
+    'push--left': css`
+      ${logicalCSS('border-left', 'none')}
+      ${logicalCSS('border-right', euiTheme.border.thick)}
+    `,
+
+    // Padding
+    paddingSizes: {
+      none: css`
+        ${composeFlyoutPadding(euiThemeContext, paddingSize)}
+      `,
+      s: css`
+        ${composeFlyoutPadding(euiThemeContext, paddingSize)}
+      `,
+      m: css`
+        ${composeFlyoutPadding(euiThemeContext, paddingSize)}
+      `,
+      l: css`
+        ${composeFlyoutPadding(euiThemeContext, paddingSize)}
+      `,
+    },
```

Comment:
Actually sorry, hang on a sec. Why are you passing `paddingSize` into this the `euiFlyoutStyles()` Emotion function? You should be statically composing the various sizes like so:

```suggestion
    // Padding
    paddingSizes: {
      none: css`
        ${composeFlyoutPadding(euiThemeContext, 'none')}
      `,
      s: css`
        ${composeFlyoutPadding(euiThemeContext, 's')}
      `,
      m: css`
        ${composeFlyoutPadding(euiThemeContext, 'm')}
      `,
      l: css`
        ${composeFlyoutPadding(euiThemeContext, 'l')}
      `,
    },
```